slatedb/src/tablestore.rs

```diff
@@ -27,12 +27,14 @@ use crate::types::RowEntry;
 use crate::{blob::ReadOnlyBlob, block::Block};
 
 pub struct TableStore {
-    object_store: Arc<dyn ObjectStore>,
+    main_object_store: Arc<dyn ObjectStore>,
```

Comment:
It might be good to encapsulate these into some type like:
```
enum ObjectStoreType {
    Wal,
    Main
}

struct ObjectStores {
    main: Arc<dyn ObjectStore>,
    wal: Arc<dyn ObjectStore>
}

impl ObjectStore {
    pub(crate) object_store_for(&self, os_type: ObjectStoreType) {...}
    pub(crate) object_store_for_sst(&self, id: SsTableId) {..}
}
```

just to force the caller to explicitly specify the object store type they're using. It seems really easy to just pick the wrong object store type by mistake.